source/sdp_deserializer.c

```diff
@@ -65,49 +74,85 @@ SdpResult_t SdpDeserializer_GetNext( SdpDeserializerContext_t * pCtx,
 
     if( result == SDP_RESULT_OK )
     {
-        if( pCtx->pStart[ pCtx->currentIndex + 1 ] != '=' )
+        /* Check for overflow before accessing array */
+        if( sdpADD_WILL_OVERFLOW( pCtx->currentIndex, 1 ) )
+        {
+            result = SDP_RESULT_MESSAGE_MALFORMED;
+        }
+        else if( pCtx->currentIndex + 1 >= pCtx->totalLength )
+        {
+            result = SDP_RESULT_MESSAGE_MALFORMED_NOT_ENOUGH_INFO;
+        }
+        else if( pCtx->pStart[ pCtx->currentIndex + 1 ] != '=' )
         {
             result = SDP_RESULT_MESSAGE_MALFORMED_EQUAL_NOT_FOUND;
         }
     }
 
     if( result == SDP_RESULT_OK )
     {
-        for( i = pCtx->currentIndex + 2; i < pCtx->totalLength; i++ )
+        /* Check for overflow before loop start */
+        if( sdpADD_WILL_OVERFLOW( pCtx->currentIndex, 2 ) )
         {
-            if( pCtx->pStart[ i ] == '\n' )
+            result = SDP_RESULT_MESSAGE_MALFORMED;
+        }
+        else
+        {
+            for( i = pCtx->currentIndex + 2; i < pCtx->totalLength; i++ )
             {
-                break;
+                if( pCtx->pStart[ i ] == '\n' )
+                {
+                    break;
+                }
             }
-        }
 
-        if( i == pCtx->totalLength )
-        {
-            /* No '\n' found. */
-            result = SDP_RESULT_MESSAGE_MALFORMED_NEWLINE_NOT_FOUND;
+            if( i == pCtx->totalLength )
+            {
+                /* No '\n' found. */
+                result = SDP_RESULT_MESSAGE_MALFORMED_NEWLINE_NOT_FOUND;
+            }
         }
 
         if( result == SDP_RESULT_OK )
         {
-            if( pCtx->pStart[ i - 1 ] == '\r' )
-            {
-                valueLength = i - pCtx->currentIndex - 3;
-            }
-            else
+            /* Calculate value length based on line ending */
+            size_t offset = ( i > 0 && pCtx->pStart[ i - 1 ] == '\r' ) ? 3 : 2;
```

Comment:
The condition seems incorrect when `i>0`. That seems impossible from previous loop. Can we just skip this check `i>0`?